src/language/branches.md

```diff
@@ -0,0 +1,295 @@
+      print("$a and $b match");    
+  }
+  ```  
+- **Variable patterns**: the matched value binds to a new variable.
+  ```dart
+  switch ((1, 2)) {
+    case (var a, var b): ...     // Matches if 1 and 2 can be assigned to a and b
+  }
+  ```  
+- **Logical** or **relational patterns**: the object matches if the comparison to a constant returns `true`.
+  ```dart
+  switch (code) {
+    case >= first && <= last: ...    
+    // Matches if code's value is between the constant values of first and last
+  }
+  ```  
+- **Null check** or **assert patterns**: the match is first on whether the object is not null, then on the value.
+  ```dart
+  switch (maybeString) {
+    case var s?:                 
+    // Matches if maybe is not null, and then against the inner pattern "var s"
+  }
+  ```  
+- **Object patterns**: the match exposes getters on the object type.
+  ```dart
+  switch (Shape shape) {
+    case Rect(width: var w, height: var h): ...
+    // Matches if shape is of type Rect, and then against the properties of Rect
+  }
+  ```  
+
+Values destructured by a case pattern become local variables, only available in the
+scope of that clause. 
+
+### Switch statements
+
+Non-empty `case` clauses implicitly jump to the end of the switch after completion.
+Other valid ways to end a non-empty `case` clause are a [`continue`][break],
+[`throw`][], or [`return`][] statement.
+
+Use a `default` or wildcard `_` clause to execute code when no `case` clause matches:
+
+```dart
+var command = 'OPEN';
+switch (command) {
+  case 'CLOSED':
+    executeClosed();
+  case 'PENDING':
+    executePending();
+  case 'APPROVED':
+    executeApproved();
+  case 'DENIED':
+    executeDenied();
+  case 'OPEN':
+    executeOpen();
+  default:
+    executeUnknown();
+}
+```
+
+Empty cases fall through to the next case. For an empty case that does not fall
+through, use [`break`][break] for its body:
+
+```dart
+var command = 'CLOSED';
+switch (command) {
+  case 'CLOSED': // Empty case falls through.
+  case 'NOW_CLOSED':
+    // Runs for both CLOSED and NOW_CLOSED.
+    executeNowClosed();
+}
+```
+
+For non-sequential fall-through, you can use a [`continue` statement][break] and
+a label:
+
+```dart
+var command = 'CLOSED';
+switch (command) {
+  case 'CLOSED':
+    executeClosed();
+    continue nowClosed;
+  // Continues executing at the nowClosed label.
+
+  nowClosed:
+  case 'NOW_CLOSED':
+    // Runs for both CLOSED and NOW_CLOSED.
+    executeNowClosed();
+}
+```
+
+### Switch expressions
+
+_Switch expressions_ write a result value determined by a switch
+directly into a variable or return statement. 
+
+The syntax rules for a switch expression are:
+- The expression body is a series of case patterns.
+- Cases _do not_ use the `case` keyword.
+- Case patterns are followed by `=>`, and then a case body. 
+- Case bodies terminate with a comma `,`.
+- The switch expression body terminates with a semicolon `;`.
```

Comment:
That's not a property of the switch expression or its body, it is a requirement in some situations of the enclosing construct: A statement consisting of an expression ends in a semicolon (e.g., `cond ? 1 : switch (x) {...};` needs the semicolon because it's a statement of the form `e;` where `e` is an expression). In other situations the `}` could be followed by a `,`, say, if the switch is an actual argument in a function invocation.

So, maybe:
```suggestion
- The switch expression body may be followed by a semicolon `;`, e.g., if it is part of a declaration or a statement.
```

Reply:
Removed the line entirely in favor of just saying it can be used anywhere an expression can be